OmsAgent/omsagent.py

```diff
@@ -692,6 +735,51 @@ def get_vmresourceid_from_metadata():
         hutil_log_error('Unexpected error from Metadata service')
         return None
 
+def get_azure_region_from_imds():
+    req = urllib.request.Request(VMResourceIDMetadataEndpoint)
+    req.add_header('Metadata', 'True')
+
+    try:
+        response = json.loads(urllib.request.urlopen(req).read())
+
+        if ('compute' not in response or response['compute'] is None):
+            return None #classic vm
+
+        if ('location' not in response['compute'] or response['compute']['location'] is None):
+            return None #classic vm
+
+        return response['compute']['location']
+    except urllib.error.HTTPError as e:
+        hutil_log_error('Request to Metadata service URL ' \
+                        'failed with an HTTPError: {0}'.format(e))
+        hutil_log_info('Response from Metadata service: ' \
+                       '{0}'.format(e.read()))
+        return None
+    except:
+        hutil_log_error('Unexpected error from Metadata service')
+        return None
+
+def get_azure_cloud_domain():
+    try:
+        location = get_azure_region_from_imds()
+        # look for the cloud type associated with current location
+        for cloudName, data in CloudRegionMap.items():
+            domain = data['Domain']
+            regions = data['Regions']
+            for region in regions:
+                regionCompacted = region.replace(" ", "")
+                if regionCompacted.lower() == location.lower():
+                    hutil_log_info("Detecting cloud type '%s' by using the current location '%s' retrieved from IMDS. The domain '%s' will be used." % (cloudName, location, domain))
+                    return domain
+
+        defaultDomain = CloudRegionMap[DefaultCloudName]['Domain']
+        hutil_log_info("Warning: We couldn't find a cloud type associated with the current location '%s', fallback to '%s' domain '%s'" % (location, DefaultCloudName, defaultDomain))
+        return defaultDomain
+    except Exception as e:
```

Comment:
In this case we should also fall back to `DefaultCloudName`

Reply:
that what I did

Reply:
I'm talking about the exception. If an exception is thrown, we just return None.